src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -20,7 +20,10 @@ public override async Task Invoke(IIncomingLogicalMessageContext context, Func<I
     {
         ValidateTransactionMode(context);
 
-        using var storageSession = context.Builder.GetService<ICompletableSynchronizedStorageSession>() ?? NoOpCompletableSynchronizedStorageSession.Instance;
+#pragma warning disable CA2007
+        await using var storageSession = context.Builder.GetService<ICompletableSynchronizedStorageSession>()
```

Comment:
Super award but you should configure the awaiter which influences the finally block https://github.com/dotnet/csharplang/discussions/2661